gateway/gateway.go

```diff
@@ -308,22 +309,25 @@ func (gateway *HandleT) dbWriterWorkerProcess(process int) {
 		for _, userWorkerBatchRequest := range breq.batchUserWorkerBatchRequest {
 			jobList = append(jobList, userWorkerBatchRequest.jobList...)
 		}
-		err := gateway.jobsDB.WithStoreSafeTx(func(tx jobsdb.StoreSafeTx) error {
-			if gwAllowPartialWriteWithErrors {
-				errorMessagesMap = gateway.jobsDB.StoreWithRetryEachInTx(tx, jobList)
-			} else {
-				err := gateway.jobsDB.StoreInTx(tx, jobList)
-				if err != nil {
-					gateway.logger.Errorf("Store into gateway db failed with error: %v", err)
-					gateway.logger.Errorf("JobList: %+v", jobList)
-					return err
+
+		err := misc.RetryWith(context.Background(), gateway.jobdDBRequestTimeout, gateway.jobdDBMaxRetries, func(ctx context.Context) error {
+			return gateway.jobsDB.WithStoreSafeTx(func(tx jobsdb.StoreSafeTx) error {
+				if gwAllowPartialWriteWithErrors {
+					errorMessagesMap = gateway.jobsDB.StoreWithRetryEachInTx(ctx, tx, jobList)
```

Comment:
we need to respect the context inside StoreWithRetryEachInTx, so most probably the method's return signature must include an error alongside the errorMap